model/src/application/interaction/application_command/option.rs

```diff
@@ -318,6 +343,18 @@ pub enum CommandOptionValue {
     Attachment(Id<AttachmentMarker>),
     Boolean(bool),
     Channel(Id<ChannelMarker>),
+    /// Value of a focused field.
+    ///
+    /// Since Discord does not validate focused fields, they are sent as strings.
+    /// This means that you will not necessarily get a valid number from number options.
+    ///
+    /// See [Discord Docs/Autocomplete].
+    ///
+    /// The actual [`CommandOptionType`] is available through the second tuple value.
+    ///
+    /// [Discord Docs/Autocomplete]: https://discord.com/developers/docs/interactions/application-commands#autocomplete
+    /// [`CommandOptionType`]: crate::application::command::CommandOptionType
+    Focused(String, CommandOptionType),
```

Comment:
The problem with this is we now have a variant of `CommandOptionValue` that isn't actually in Discord's API. I think I would prefer to fix this bug by keeping `ApplicationCommandAutocomplete` a separate struct, especially since most of the time applications deal with autocomplete separately anyway. I think this PR needs to be reconsidered.

Reply:
I thought of this too.
My 2 arguments of this decision were:
1. Keeping a separate struct would mean that we need to repeat 100% of the code just for having one additional enum variant.
2. I personally think that `focused: true` is also some kind of additional type to the available ones. Which modifies the behavior of `value`.

Reply:
I personally value matching the API as closely as we can over code duplication concerns, but others should weigh in. If I have some time I will see what it's like to fix it keeping the Autocomplete variant.

Reply:
From a user perspective, I'd prefer having one struct with an extra type. I understand the variant may not natually exist in the API, but if you look at it from a data perspective, a piece of data with an extra value "focused" could be considered separate from the same data without. Thus, representing them as two separate entities would not be a problem imo.

Reply:
I also find that adding another variant is better, it makes easier to implement functions which does some processing with incoming interactions without handling both cases.